packages/template-cli/src/index.ts

```diff
@@ -1,19 +1,89 @@
 #!/usr/bin/env node
+import yargs from 'yargs'
 import { basename, resolve } from 'path'
-import { ExitMessage } from './domain/model/cli_response'
-import { runCli } from './domain/prompt'
+import { ExitMessage, CliOptions } from './domain/model/cli_response'
+import { runCli, runConfig, generateSampleConfig } from './domain/prompt'
 import chalk from 'chalk'
 
-// main cli entry
-(async () => {
-    const default_project_name = basename(resolve(process.cwd()))
-    const args = process.argv
+
+async function cliCommand(argv: CliOptions) {
     try {
-        const response: ExitMessage = await runCli(default_project_name, args.slice(2))
+        const default_project_name = basename(resolve(process.cwd()));
+        // let argv = giveMeArgs()
+        let response: ExitMessage = <ExitMessage>{}
+
+        if (argv.configfile) {
+            // run with a config file
+            response = await runConfig(argv)
+        } else if (argv.generatesampleconfig) {
+            // generate sample config without value replaced
+            response = await generateSampleConfig()
+        } else if (argv.interactive) {
+            // run cli flow
+            response = await runCli(default_project_name, argv)
+        } else {
+            console.log(chalk.cyan(yargs.showHelp())) //"Please select an appropriate option"
+            return process.exit(0)
+        }
         console.log(chalk.cyan(response.message))
         return process.exit(0)
     } catch (ex) {
-        console.log(ex.message)
+        let exCaught = ex as ExitMessage
+        console.log(chalk.red(exCaught.code))
```

Comment:
One log?

Reply:
it's better to log out the message and code separately I think - as opposed to concat'ing them together no?